src/topology/continuous_function/units.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import topology.continuous_function.compact
+import analysis.normed_space.units
+import algebra.algebra.spectrum
+
+/-!
+# Units of continuous functions
+
+This file concerns itself with `C(α, β)ˣ` and `C(α, βˣ)` when `α` is a topological space
+and `β` is a normed ring.
+-/
+
+namespace continuous_map
+
+section units
+
+section normed_ring
+
+variables {α : Type*} [topological_space α] {β : Type*} [normed_ring β]
+
+/-- Equivalence between continuous maps into the units of a normed ring the
+the units of the ring of continuous functions. -/
+@[simps]
+def units_lift : C(α, βˣ) ≃ C(α, β)ˣ :=
+{ to_fun := λ f,
+  { val := ⟨coe ∘ f, units.continuous_coe.comp f.continuous⟩,
+    inv := ⟨λ x, ↑(f x)⁻¹, units.continuous_coe.comp (continuous_inv.comp f.continuous)⟩,
+    val_inv := by { ext, simp only [coe_mul, coe_mk, pi.mul_apply, units.mul_inv, coe_one,
+      pi.one_apply] },
+    inv_val := by { ext, simp only [coe_mul, coe_mk, pi.mul_apply, units.inv_mul, coe_one,
+      pi.one_apply]} },
+  inv_fun := λ f,
+  { to_fun := λ x, ⟨f x, f⁻¹ x, (f.val.coe_mul f.inv ▸ continuous_map.congr_fun f.val_inv x),
+                                (f.inv.coe_mul f.val ▸ continuous_map.congr_fun f.inv_val x)⟩,
+    continuous_to_fun := continuous_induced_rng (continuous.prod_mk (f : C(α, β)).continuous
+      $ mul_opposite.continuous_op.comp (continuous_map.continuous (f⁻¹ : C(α, β)ˣ))) },
+  left_inv := λ f, by { ext, refl },
+  right_inv := λ f, by { ext, refl } }
```

Comment:
Slightly golfed:
```suggestion
{ to_fun := λ f,
  { val := ⟨λ x, ↑(f x), units.continuous_coe.comp f.continuous⟩,
    inv := ⟨λ x, ↑(f x)⁻¹, units.continuous_coe.comp f.continuous.inv⟩,
    val_inv := ext $ λ a, units.mul_inv _,
    inv_val := ext $ λ a, units.inv_mul _ },
  inv_fun := λ f,
  { to_fun := λ x, ⟨f x, f⁻¹ x, continuous_map.congr_fun f.mul_inv x,
                                continuous_map.congr_fun f.inv_mul x⟩,
    continuous_to_fun := continuous_induced_rng $
      (f : C(α, β)).continuous.prod_mk $
        mul_opposite.continuous_op.comp (↑f⁻¹ : C(α, β)).continuous },
  left_inv := λ f, by { ext, refl },
  right_inv := λ f, by { ext, refl } }
```

Note that `coe ∘ f` generates an uglier lemma through `simps` than `λ x, f x` does.

Reply:
thanks, done.